tests/phpunit/CRM/Apilimitsettings/Event/Listener/ChangeApiSettingTest.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+/**
+ * Tests the listener that changes the API settings.
+ *
+ * @group headless
+ */
+class CRM_Apilimitsettings_Event_Listener_ChangeApiSettingTest extends BaseHeadlessTest {
+
+  /**
+   * Tests the new default limit is set.
+   */
+  public function testNewDefaultApiLimitIsSet(): void {
+    $expectedContacts = 30 + civicrm_api3('Contact', 'getcount');
+    $this->createTestContacts(30);
+
+    $contacts = civicrm_api3('Contact', 'get');
+
+    $this->assertEquals($expectedContacts, $contacts['count']);
+    $this->assertCount($expectedContacts, $contacts['values']);
+  }
+
+  /**
+   * Tests specified value on query is respected.
+   */
+  public function testLimitValueOnQueryIsRespected(): void {
+    $expectedContacts = 30 + civicrm_api3('Contact', 'getcount');
+    $this->createTestContacts(30);
+
+    $contacts = civicrm_api3('Contact', 'get', [
+      'options' => ['limit' => 10],
+    ]);
+    $totalContacts = civicrm_api3('Contact', 'getcount');
+
+    $this->assertEquals(10, $contacts['count']);
+    $this->assertCount(10, $contacts['values']);
+    $this->assertEquals($expectedContacts, $totalContacts);
```

Comment:
Should line 35 & 36 runs before ` $totalContacts = civicrm_api3('Contact', 'getcount');` so it is easier to read?  Like, we should group assertion methods after the API was called to check the result?  This applies to other tests too.

Reply:
Done